src/log.cpp

```diff
@@ -80,8 +91,24 @@ void log_to_file(ru_severity min_sev, const char* file_path) {
   get_single_logger()->log_to_file(min_sev, file_path);
 }
 
+#ifdef RS2_USE_ANDROID_BACKEND
+void android_log_msg(ru_severity sev, const std::string& msg) {
+      switch (sev) {
+        case RU_SEVERITY_DEBUG:   ANDROID_LOG_WARNING("%s",msg.c_str()); break;
+        case RU_SEVERITY_INFO:    ANDROID_LOG_INFO("%s",msg.c_str()); break;
+        case RU_SEVERITY_WARNING: ANDROID_LOG_WARNING("%s",msg.c_str()); break;
+        case RU_SEVERITY_ERROR:   ANDROID_LOG_ERROR("%s",msg.c_str()); break;
+        case RU_SEVERITY_FATAL:   ANDROID_LOG_FATAL("%s",msg.c_str()); break;
+        default:;
+      }
+    }
+#endif
+
 void log_msg(ru_severity sev, const std::string& msg) {
+#ifdef RS2_USE_ANDROID_BACKEND
+  android_log_msg(sev,msg);
```

Comment:
This can come as part of further Android rework, when/if we do the work to do all of this properly.

Reply:
After our discussion yesterday I had a better look.

The code in usbhost.c which was noted (the `#ifdef DEBUG`) is better off completely stripped out at the moment - the inner true branch (when DEBUG is 1 and USE_LIBLOG is defined) does not compile for us, due to the appropriate log.h header missing. The log header provided by the NDK does not provide the ALOGD macro - I presume this is a header that comes with the AOSP rather than the NDK.

The fact we have to use the usbhost source and header here rather than the libusthost library available on the Android device is rather unfortunate and I'd rather spend the time achieving that. I'll add that to the TODO list.

If there are other places where logging is disabled behind a preprocessor directive let me know and I'll fix them.